internal/juju/client.go

```diff
@@ -225,31 +230,72 @@ func (sc *sharedClient) WaitForResource() bool {
 	return sc.waitForResources
 }
 
+// GetExternalControllerConn returns a juju connection for an external controller
+// configuration previously added to the sharedClient.
+func (sc *sharedClient) GetExternalControllerConn(name string) (api.Connection, error) {
+	sc.lockExtCtlrMap.RLock()
+	controllerConfig, ok := sc.externalControllerConfigs[name]
+	sc.lockExtCtlrMap.RUnlock()
+	if !ok {
+		return nil, errors.NotFoundf("external controller configuration for %q", name)
+	}
+	return sc.getConn(connector.SimpleConfig{
+		ControllerAddresses: controllerConfig.ControllerAddresses,
+		Username:            controllerConfig.Username,
+		Password:            controllerConfig.Password,
+		CACert:              controllerConfig.CACert,
+	})
+}
+
+// AddExternalControllerConf adds an external controller configuration
+// to the sharedClient.
+func (sc *sharedClient) AddExternalControllerConf(name string, conf ControllerConfiguration) error {
+	sc.lockExtCtlrMap.Lock()
+	sc.externalControllerConfigs[name] = conf
+	sc.lockExtCtlrMap.Unlock()
```

Comment:
Shouldn't we add it to the map after we test the connection? Not that it matters much since an error will cause us to exit early.